### PR TITLE
Parachain MMR Verification Fix

### PR DESCRIPTION
**Pull request by alistair-singh**
Adding logging and debugging commands.

usage from inside the relayer folder:
```
./build/snowbridge-relay beefy-proof -u wss://rococo-rpc.polkadot.io \
    --beefy-block-hash db058872ab692647dbaf91d4da28a22454fd76e1babd2148d02d418a0ff8adef \
    --relaychain-block 1538916 --parachain-block 63958 --parachain-id 3016
```

Investigation in ticket.
Resolves: SNO-320

### CODE REVIEW

**Review by vgeddes — COMMENTED** [review]
Looks good! Added some feedback. Mostly about error handling, etc.

Also I think we should add some toplevel commentary to `BeefyListener.generateProof` describing roughly what it does: Generating a proof for an MMR leaf, and then generating a proof for our parachain header, which should be verifiable against the parachains root in the mmr leaf.

It would also be good to link to the Polkadot code that generates the parachains root: https://github.com/paritytech/polkadot/blob/2eb7672905d99971fc11ad7ff4d57e68967401d2/runtime/rococo/src/lib.rs#L700

File: relayer/main.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
[the remark this answers is not in the document]
Comment (alistair-singh) [reply relayer/main.go:8]: Will do, I think this was gofmt on save.

**Comment by alistair-singh** [comment]
I have added the requested comments. This is ready for review.

**Review by vgeddes — APPROVED** [review]
Thanks!

**Pull request merged**